texar/modules/berts/bert_utils.py

```diff
@@ -0,0 +1,290 @@
+                    'layer_\\d+/output/LayerNorm',
+                    lambda x: x.group(0).replace('output/LayerNorm',
+                                                 'ffn/LayerNorm'),
+                    check_name_scope)
+            if model_name == check_name_scope:
+                model_name = re.sub(
+                    'layer_\\d+/intermediate/dense',
+                    lambda x: x.group(0).replace('intermediate/dense',
+                                                 'ffn/intermediate'),
+                    check_name_scope)
+            if model_name == check_name_scope:
+                model_name = re.sub('attention/output/dense',
+                                    'attention/self/output', check_name_scope)
+            if model_name == check_name_scope:
+                model_name = check_name_scope.replace(
+                    'attention/output/LayerNorm', 'output/LayerNorm')
+
+            if model_name in name_to_variable.keys():
+                assignment_map[check_name] = model_name
+                initialized_variable_names[model_name] = 1
+                initialized_variable_names[model_name + ":0"] = 1
+            else:
+                tf.logging.info('model name:{} not exist'.format(model_name))
+
+    return assignment_map, initialized_variable_names
+
+
+def init_bert_checkpoint(init_checkpoint_dir, scope_name):
+    """
+    Initializes BERT model parameters from a checkpoint.
+    Provided by Google AI Language Team.
+
+    Args:
+        init_checkpoint_dir (str): path to the checkpoint.
+        scope_name: variable scope of bert encoder.
+    """
+    tvars = tf.trainable_variables()
+    init_checkpoint = os.path.join(init_checkpoint_dir, 'bert_model.ckpt')
+    if init_checkpoint:
+        (assignment_map, initialized_variable_names
+        ) = _get_assignment_map_from_checkpoint(
+            tvars, init_checkpoint, scope_name)
+        tf.train.init_from_checkpoint(init_checkpoint, assignment_map)
+
+
+def _default_download_dir():
+    """
+    Return the directory to which packages will be downloaded by default.
+    """
+    package_dir = os.path.dirname(os.path.dirname(
+        os.path.dirname(os.path.dirname(__file__))))
+    if os.access(package_dir, os.W_OK):
+        texar_download_dir = os.path.join(package_dir, 'texar_download')
+    else:
+        # On Windows, use %APPDATA%
+        if sys.platform == 'win32' and 'APPDATA' in os.environ:
+            home_dir = os.environ['APPDATA']
+
+        # Otherwise, install in the user's home directory.
+        else:
+            home_dir = os.path.expanduser('~/')
+            if home_dir == '~/':
+                raise ValueError("Could not find a default download directory")
+
+        texar_download_dir = os.path.join(home_dir, 'texar_download')
+
+    if not os.path.exists(texar_download_dir):
+        os.mkdir(texar_download_dir)
+
+    return os.path.join(texar_download_dir, 'bert')
+
+
+def _http_get(url, cache_dir):
```

Comment:
Can [texar.data.maybe_download](https://texar.readthedocs.io/en/latest/code/data.html#texar.data.maybe_download) be used instead? If so, we may move `texar.data.maybe_download` to `texar/utils/utils_io.py`, and update all places in `examples/` where `maybe_download` is used